components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -935,6 +935,14 @@ private HashMap<String, String> getAccountStateClaims(UserRecoveryData userRecov
             userClaims.put(IdentityRecoveryConstants.ACCOUNT_LOCKED_CLAIM, Boolean.FALSE.toString());
         }
 
+        // If the recovery scenario is ASK_PASSWORD, it indicates a password set flow.
+        // Hence, update the thread-local properties to specify that this is a password set flow.
+        if (RecoveryScenarios.ASK_PASSWORD.equals(recoveryScenario)) {
+            IdentityUtil.threadLocalProperties.get().put(AccountConstants.PASSWORD_SET_FLOW, true);
+        } else {
+            IdentityUtil.threadLocalProperties.get().put(AccountConstants.PASSWORD_SET_FLOW, false);
```

Comment:
Can't see any place where clearing the threadlocal properties. 
Since threadlocal usage is bit unsafe can we go with adding a runtime claim to userClaims list and get it should in the account lock handler?
I hope we can go with a runtime claim which is not actually registered into the claims of the tenant.